### PR TITLE
add a lighter way to check whether a block exists in the Bigtable

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

#28166 

#### Summary of Changes

- add does_row_key_exist
- add does_confirmed_block_exist

### CODE REVIEW

File: storage-bigtable/src/bigtable.rs [hunk 1]
```diff
@@ -458,6 +458,31 @@ impl<F: FnMut(Request<()>) -> InterceptedRequestResult> BigTable<F> {
         Ok(rows.into_iter().map(|r| r.0).collect())
     }
 
+    /// Check wether a row key exist in the Bigtable
+    pub async fn does_row_key_exist(&mut self, table_name: &str, row_key: RowKey) -> Result<bool> {
```
Comment (CriesofCarrots) [inline storage-bigtable/src/bigtable.rs:462]:
```suggestion
    pub async fn row_key_exists(&mut self, table_name: &str, row_key: RowKey) -> Result<bool> {
```

File: storage-bigtable/src/lib.rs [hunk 2]
```diff
@@ -535,6 +535,22 @@ impl LedgerStorage {
         })
     }
 
+    /// Does the confirmed block exist in the Bigtable
+    pub async fn does_confirmed_block_exist(&self, slot: Slot) -> Result<bool> {
+        debug!(
+            "LedgerStorage::does_confirmed_block_exist request received: {:?}",
```
Comment (CriesofCarrots) [inline storage-bigtable/src/lib.rs:541]:
```suggestion
            "LedgerStorage::confirmed_block_exists request received: {:?}",
```

File: storage-bigtable/src/bigtable.rs (hunk 1 above)
Comment (CriesofCarrots) [reply storage-bigtable/src/bigtable.rs:462]: etc...

**Review by CriesofCarrots — COMMENTED** [review]
Code looks good, just nitpicking method names

File: storage-bigtable/src/bigtable.rs [hunk 3]
```diff
@@ -458,6 +458,31 @@ impl<F: FnMut(Request<()>) -> InterceptedRequestResult> BigTable<F> {
         Ok(rows.into_iter().map(|r| r.0).collect())
     }
 
+    /// Check wether a row key exist in the Bigtable
```
[the remark this answers is not in the document]
Comment (yihau) [reply storage-bigtable/src/bigtable.rs:461]: oh! haha 🐑 Thank you!

File: storage-bigtable/src/lib.rs [hunk 4]
```diff
@@ -535,6 +535,22 @@ impl LedgerStorage {
...
+            slot
+        );
+        inc_new_counter_debug!("storage-bigtable-query", 1);
+        let mut bigtable = self.connection.client();
+
+        let does_block_exist = bigtable
+            .row_key_exists("blocks", slot_to_blocks_key(slot))
+            .await?;
+
+        Ok(does_block_exist)
```
Comment (CriesofCarrots) [inline storage-bigtable/src/lib.rs:551]:
```suggestion
        let block_exists = bigtable
            .row_key_exists("blocks", slot_to_blocks_key(slot))
            .await?;

        Ok(block_exists)
```

**Review by CriesofCarrots — APPROVED** [review]
One last tiny nit

**Pull request merged**